ehsm_kms_service/delete_secret_thread.js

```diff
@@ -0,0 +1,137 @@
+        init_secret_delete_thread(DB)
+    } catch (error) {
+        console.error('error removeTimer :: ', error)
+    }
+}
+
+/**
+ * Expose the interface externally, scheduled tasks of delete secret
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @param {Object} plannedDeleteTime : planned delete time for secret eg. '1659931644416'
+ * @returns
+ */
+const startDeleteSecretTask = (DB, appid, secretName, plannedDeleteTime) => {
+    try {
+        let delayMillisecond = plannedDeleteTime - new Date().getTime()
+        let needAdd = true
+        // process exist timer, 
+        // If the new time is within plus or minus 10 seconds of the original time, keep the original plan unchanged; otherwise, clear the original plan and add again
+        if (exist_timer.hasOwnProperty(secretName)) {
+            if (exist_timer[secretName].plannedDeleteTime + 10000 > plannedDeleteTime && exist_timer[secretName].plannedDeleteTime - 10000 < plannedDeleteTime) {
+                needAdd = false
+            } else {
+                clearTimeout(exist_timer[secretName].timer)
+            }
+        }
+        //If the planned delete time exceeds 12 hours, the thread will not be started
+        if (delayMillisecond > 12 * 1000 * 60 * 60) {
+            needAdd = false
+        }
+        if (needAdd) {
+            //Force delete according to planned deletion time
+            let timer = setTimeout(_deleteSecret, delayMillisecond, DB, appid, secretName)
+            exist_timer[secretName] = {
+                timer,
+                plannedDeleteTime
+            }
+        }
+    } catch (error) {
+        console.error('error deleteSecret :: ', error)
+    }
+}
+
+/**
+ * Delete secret from the database by secretName
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @returns
+ */
+const _deleteSecret = async (DB, appid, secretName) => {
+    try {
+        const forceDeleteData = require('./secret_manager_apis').forceDeleteData
+        if (await forceDeleteData(DB, appid, secretName)) {
+            console.info(`delete secret thread :: The ${base64_decode(secretName)} delete success`)
+        } else {
+            console.error(`delete secret thread :: The ${base64_decode(secretName)} delete failed`)
+        }
+    } catch (error) {
+        console.error('error _deleteSecret :: ', error)
+    }
+    // remove exist timer
+    removeTimer(secretName, DB)
+}
+
+/**
+ * Initialize the delete thread and call it when the server starts
+ * @param {Object} DB : database control
+ * @returns
+ */
+const init_secret_delete_thread = (DB) => {
+    if (init_interval == undefined || init_interval._destroyed) {
+        //Execute _initTimer every six hours
+        init_interval = setInterval(_initTimer, 1000 * 60 * 60 * 6, DB)
```

Comment:
setInterval 6h, will run the thread @0h, 6h, 12h... OR @6h, 12h,18h ... ?  Which one?

Reply:
6H, 12h,18h

Reply:
seems you can combine the _initTimer and init_secret_delete_thread into one function, which can be named as _secret_delete_timer